src/libmongoc/src/mongoc/mongoc-crypto-cng.c

```diff
@@ -141,6 +143,101 @@ _mongoc_crypto_cng_hmac_or_hash (
    return retval;
 }
 
+static int
+_crypto_hash_size (mongoc_crypto_t *crypto)
+{
+   if (crypto->algorithm == MONGOC_CRYPTO_ALGORITHM_SHA_1) {
+      return MONGOC_SCRAM_SHA_1_HASH_SIZE;
+   } else if (crypto->algorithm == MONGOC_CRYPTO_ALGORITHM_SHA_256) {
+      return MONGOC_SCRAM_SHA_256_HASH_SIZE;
+   }
+   return 0;
+}
+
+/* Wrapper for BCryptDeriveKeyPBKDF2 */
+#if defined(MONGOC_HAVE_BCRYPT_PBKDF2)
+static int
+_bcrypt_derive_key_pbkdf2 (BCRYPT_ALG_HANDLE prf,
+                           const char *password,
+                           size_t password_len,
+                           const uint8_t *salt,
+                           size_t salt_len,
+                           uint32_t iterations,
+                           size_t output_len,
+                           unsigned char *output)
+{
+   // Make non-const versions of password and salt.
+   char *password_copy = bson_strndup (password, password_len);
+   char *salt_copy = bson_strndup (salt, salt_len);
+
+   NTSTATUS status = BCryptDeriveKeyPBKDF2 (prf,
+                                            (unsigned char *) password_copy,
+                                            password_len,
+                                            (unsigned char *) salt_copy,
+                                            salt_len,
+                                            iterations,
+                                            output,
+                                            output_len,
+                                            0);
+   if (!NT_SUCCESS (status)) {
+      MONGOC_ERROR ("BCryptDeriveKeyPBKDF2(): %ld", status);
+   }
+   bson_free (password_copy);
+   bson_free (salt_copy);
+   return (int) status;
+}
+#endif
+
+/* Compute the SCRAM step Hi() as defined in RFC5802 */
+static void
+mongoc_crypto_cng_derive_key_pbkdf2 (mongoc_crypto_t *crypto,
```

Comment:
```suggestion
_bcrypt_derive_key_pbkdf2 (mongoc_crypto_t *crypto,
```

Suggest using the same name as the alternative that uses `BCryptDeriveKeyPBKDF2` and wrap in `#else` / `#endif` since this is only called when the alternative is unavailable.